storage/src/mmr/journaled.rs

```diff
@@ -1,32 +1,54 @@
 //! An MMR backed by a fixed-item-length journal.
+//!
+//! A journal is used to store all unpruned MMR nodes, and a metadata store is used to preserve each
```

Comment:
nit: You can link to journal with `[commonware_strorage::journal]` (same goes for metadata)

Reply:
That didn't seem to resolve, but [`crate::journal`] did, so changed to that.